src/transactions/InvokeHostFunctionOpFrame.cpp

```diff
@@ -700,20 +593,24 @@ InvokeHostFunctionOpFrame::doApply(Application& app, AbstractLedgerTxn& ltx,
     }
 
     maybePopulateDiagnosticEvents(cfg, out, metrics);
-    // This may throw, but only in case of the Core version misconfiguration.
-    int64_t rentFee = rust_bridge::compute_rent_fee(
-        cfg.CURRENT_LEDGER_PROTOCOL_VERSION,
-        ltx.loadHeader().current().ledgerVersion, rustEntryRentChanges,
-        sorobanConfig.rustBridgeRentFeeConfiguration(), ledgerSeq);
-    mParentTx.consumeRefundableSorobanResources(metrics.mEmitEventByte,
-                                                rentFee);
+
+    if (!mParentTx.consumeRefundableSorobanResources(
+            metrics.mEmitEventByte, out.rent_fee,
+            cfg.CURRENT_LEDGER_PROTOCOL_VERSION, sorobanConfig, cfg))
```

Comment:
This should be passing the current protocol version (`ltx.loadHeader().current().ledgerVersion`), not `cfg.CURRENT_LEDGER_PROTOCOL_VERSION` from what I can see. `TransactionFrame::computeSorobanResourceFee` calls `rust_bridge::compute_transaction_resource_fee(
        cfg.CURRENT_LEDGER_PROTOCOL_VERSION, protocolVersion, ...);`, so you'd be passing the max in place of the current.

This comment applies to the calls in the other ops as well.

Reply:
Nice catch, fixed.